snippets/csharp/System.Collections.Generic/IEqualityComparerT/Overview/program.cs

```diff
@@ -2,81 +2,69 @@
 using System;
 using System.Collections.Generic;
 
-class Example
+static class Example
 {
-   static void Main()
-   {
-      BoxEqualityComparer boxEqC = new BoxEqualityComparer();
-
-      var boxes = new Dictionary<Box, string>(boxEqC);
-
-      var redBox = new Box(4, 3, 4);
-      AddBox(boxes, redBox, "red");
+    static void Main()
+    {
+        BoxEqualityComparer comparer = new();
 
-      var blueBox = new Box(4, 3, 4);
-      AddBox(boxes, blueBox, "blue");
+        Dictionary<Box, string> boxes = new(comparer);
 
-      var greenBox = new Box(3, 4, 3);
-      AddBox(boxes, greenBox, "green");
-      Console.WriteLine();
+        AddBox(new Box(4, 3, 4), "red");
+        AddBox(new Box(4, 3, 4), "blue");
+        AddBox(new Box(3, 4, 3), "green");
 
-      Console.WriteLine("The dictionary contains {0} Box objects.",
-                        boxes.Count);
-   }
+        Console.WriteLine($"The dictionary contains {boxes.Count} Box objects.");
 
-   private static void AddBox(Dictionary<Box, String> dict, Box box, String name)
-   {
-      try {
-         dict.Add(box, name);
-      }
-      catch (ArgumentException e) {
-         Console.WriteLine("Unable to add {0}: {1}", box, e.Message);
-      }
-   }
+        void AddBox(Box box, string name)
+        {
+            try
+            {
+                boxes.Add(box, name);
+            }
+            catch (ArgumentException e)
+            {
+                Console.WriteLine($"Unable to add {box}: {e.Message}");
+            }
+        }
+    }
 }
 
-public class Box
+class Box
 {
-    public Box(int h,  int l, int w)
-    {
-        this.Height = h;
-        this.Length = l;
-        this.Width = w;
-    }
-
-    public int Height { get; set; }
-    public int Length { get; set; }
-    public int Width { get; set; }
+    public int Height { get; }
+    public int Length { get; }
+    public int Width { get; }
 
-    public override String ToString()
+    public Box(int height, int length, int width)
```

Comment:
Is it common practice to put constructor declarations below property declarations?

Reply:
In my experience yes, for stateful properties. They go alongside fields and events, which together represent the state of the type, then constructors that represent initialization of that state, then everything else. I think the idea goes back to the days of UML diagrams. But there are no rules on this.

Reply:
By stateful in this context you mean that they're backed by fields? I've never seen that ordering personally, though you're right to point out there are no rules. The introduction of primary constructors in the upcoming C# release will most likely render this concern obsolete, with the constructor being placed at the top of the class declaration.

Reply:
The docs style guide doesn't prefer either style.